common/src/main/java/tachyon/util/io/BufferUtils.java

```diff
@@ -39,8 +37,16 @@ public static void cleanDirectBuffer(ByteBuffer buffer) {
       return;
     }
     if (buffer.isDirect()) {
-      Cleaner cleaner = ((DirectBuffer) buffer).cleaner();
-      cleaner.clean();
+      try {
+        final Method getCleanerMethod = buffer.getClass().getMethod("cleaner");
+        getCleanerMethod.setAccessible(true);
+        final Object cleaner = getCleanerMethod.invoke(buffer);
+        if (cleaner != null) {
+          cleaner.getClass().getMethod("clean").invoke(cleaner);
+        }
+      } catch (Exception e) {
```

Comment:
@danielfree I think your reasoning is reasonable. I think it's fine to add `LOG.warn("Fail to unmap direct buffer due to ", e)` before `buffer = null`.